boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts

```diff
@@ -22,12 +22,47 @@
 	};
 
 	chosen {
+#if defined(CONFIG_CODE_ITCM)
 		zephyr,flash = &itcm0;
+#elif defined(CONFIG_CODE_HYPERFLASH)
+		zephyr,flash = &hyperflash0;
+#elif defined(CONFIG_CODE_QSPI)
+		zephyr,flash = &qspi0;
+#endif
 		zephyr,sram = &dtcm0;
 		zephyr,console = &uart1;
 	};
 };
 
+&flexspi0 {
+	hyperflash0: hyperflash@0 {
+		/* Cypress S26KS512SDPBHI02 */
+		reg = <0x60000000 0x4000000>;
+#if defined(CONFIG_CODE_HYPERFLASH)
+		status = "ok";
+#else
+		status = "disabled";
+#endif
+	};
+
+	qspi0: qspi@0 {
+		/* ISSI IS25WP064AJBLE */
+		reg = <0x60000000 0x800000>;
+#if defined(CONFIG_CODE_QSPI)
+		status = "ok";
+#else
+		status = "disabled";
+#endif
+	};
+};
+
+&semc0 {
+	sdram0: sdram@0 {
+		/* Micron MT48LC16M16A2B4-6AIT:G */
+		reg = <0x80000000 0x2000000>;
```

Comment:
Ok now that I realize this is dram, this should just be a normal memory node.  There's not association with the semc0.
        memory@80000000 {
                device_type = "memory";
                reg = <0x80000000 0x2000000 >; /* 32 MB */
        };

Reply:
Doesn't it still need an alias so you could choose it for `zephyr,sram`? Do the flash changes look ok?

Reply:
Sure, you can put a label on it:

mem: memory@8000...

Reply:
Done